File: examples/javascript/search-jobs.mjs
import { createGunzip } from "node:zlib";
import { Readable } from "node:stream";
import readline from "node:readline";

const url = "https://github.com/rocketlist-ai/startup-jobs/releases/latest/download/jobs.jsonl.gz";
const response = await fetch(url);
if (!response.ok || !response.body) throw new Error(`Download failed: ${response.status}`);

const lines = readline.createInterface({
  input: Readable.fromWeb(response.body).pipe(createGunzip()),
  crlfDelay: Infinity,
});

for await (const line of lines) {
  const job = JSON.parse(line);
  if (/founder|strategy|operations/i.test(job.title ?? "") && /Germany/i.test(job.country ?? "")) {
    console.log(`${job.company_name} — ${job.title} — ${job.url}`);
  }
}

File: examples/python/search_jobs.py
"""Minimal local query example: pip install pandas pyarrow."""

import pandas as pd


URL = "https://github.com/rocketlist-ai/startup-jobs/releases/latest/download/jobs.parquet"
jobs = pd.read_parquet(URL)

matches = jobs[
    jobs["title"].fillna("").str.contains("founder|strategy|operations", case=False)
    & jobs["country"].fillna("").str.contains("Germany", case=False)
]

print(matches[["company_name", "title", "location", "url"]].head(25).to_string(index=False))

